rest-service/manager_rest/plugins_update/manager.py

```diff
@@ -221,18 +222,30 @@ def get_reevaluated_plan(self, blueprint):
                                        config.instance.file_server_root)
         plan = blueprint.plan
         if not _did_plugins_to_install_change(temp_plan, plan):
-            raise PluginsUpdateError("No new plugins changes have been "
-                                     "detected when re-parsed blueprint {0}, "
-                                     "aborting update.".format(blueprint.id))
+            raise PluginsUpdateError(
+                'Found no plugins to update for "{0}" '
+                'blueprint, aborting plugins update.'.format(
+                    blueprint.id))
         return plan
 
 
 def _did_plugins_to_install_change(temp_plan, plan):
-    temp_plugins = temp_plan[DEPLOYMENT_PLUGINS_TO_INSTALL] + \
-                   temp_plan[WORKFLOW_PLUGINS_TO_INSTALL]
-    plugins = plan[DEPLOYMENT_PLUGINS_TO_INSTALL] + plan[
-        WORKFLOW_PLUGINS_TO_INSTALL]
-    name_to_plugin = {p[PLUGIN_NAME_KEY]: p for p in plugins}
+    # Maintaining backward comparability for older blueprints
+    if not plan.get(HOST_AGENT_PLUGINS_TO_INSTALL, []):
```

Comment:
note: this is only done for `plan` but not for `temp_plan`, and that is why the tests fail

probably update the `temp_plan` in the failing tests